Microsoft Azure App Services Extension beta docs (#7124)
DataDog/documentation: content/en/infrastructure/serverless/azure_app_services.md

---
title: Microsoft Azure App Services Extension
kind: documentation
further_reading:
- link: "/integrations/azure_app_services/"
  tag: "Documentation"
  text: "Azure App Service"
- link: "integrations/azure_app_service_environment"
  tag: "Documentation"
  text: "Azure App Service Environment"
---
<div class="alert alert-warning"> This service is in public beta. If you have any feedback, please <a href="/help">contact Datadog support</a>.</div>

> ## Azure App Services Overview

Reviewer: ```suggestion
## Overview
```

- .NET Framework 4.7 (AAS does not support more recent versions)
- .NET Core 2.1
- .NET Core 2.2 (Microsoft support ended 2019-12-23)
- .NET Core 3.0 (Microsoft supports will end 2020-03-03)
- .NET Core 3.1

## Installation

1. Open the [Azure Portal][7] and navigate to the dashboard for the Azure App Services instance you wish to instrument with Datadog.
2. Go to the Application settings tab of the Configuration page.
3. Add your Datadog API key as an application setting called `DD_API_KEY`. You can find or create a new API key in your [Datadog Account][8].
4. If you use the EU Datadog site (datadoghq.eu domain), add an application setting `DD_SITE` with datadoghq.eu as the value.
    By default, the extension submits data to the US Datadog site (the datadoghq.com domain). So if you use the US Datadog site, there is no additional application setting required.
5. Go to the extensions page and click **Add**.
6. Select the Datadog APM extension.
7. Accept the legal terms, click **OK**, and wait for the installation to complete.
8. Restart the main application: click **Stop**, wait for a full stop, then click **Start**.

## Logs and Trace ID Injection

> Logs for Azure Web Apps can be submitted to Datadog with Eventhub using the process described in our [Azure Integration documentation][9].

Reviewer: ```suggestion
Logs for Azure Web Apps can be submitted to Datadog with Eventhub using the process described in the [Azure Integration documentation][9].
```

Once you establish the logging pipeline for your application, Trace ID injection allows you to [connect logs and traces][10] in Datadog. To enable this with the extension, add an application setting `DD_LOGS_INJECTION:true`.

> **Note**: Trace ID injection occurs in the application, so application logs include the trace ID. Other categories of [diagnostic Logs available from Azure][11], like HTTP logs and audit logs, do not include the trace ID.

Reviewer: ```suggestion
**Note**: Trace ID injection occurs in the application, so application logs include the trace ID. Other categories of [diagnostic logs available from Azure][11], like HTTP logs and audit logs, do not include the trace ID.
```

## Troubleshooting

### 5XX errors

> If your app begins throwing 5XX errors immediately after installation, we recommend first trying to reinstall with the application in a full stop. To do this:

Reviewer: ```suggestion
If your app begins throwing 5XX errors immediately after installation, first try to reinstall with the application in a full stop. To do this:
```
